### ISSUE TITLE
stuck when change the interval of checkpoint_config

### ISSUE THREAD

**Issue by ZR19940115** [open-mmlab/mmdetection#2012]
when i change the interval from 1 to 4 or 2 of the checkpoint_config, it may stuck randomly when training. but it is ok when interval is 1 originally. do you know how to solve this bug?

**Comment by hellock** [comment]
That's weird. Which config are you using?

**Comment by ZR19940115** [comment]
> That's weird. Which config are you using?

I am using cascade_rcnn_r50_fpn_1x.py. I want to save disk space, so I change the interval from 1 to 4. then i find the training stuck the second day. it neither stopped training, nor continued training.

**Comment by ZwwWayne** [comment]
Could you provide specific logs, configs,  and running commands for us to reproduce it?
The environment information might also be helpful.

**Comment by ZR19940115** [comment]
> Could you provide specific logs, configs, and running commands for us to reproduce it?
> The environment information might also be helpful.

My classmate also meet this problem on his computer, and our environments are different, so i guess it is because of another unknown reasons. 
My running command is :
`python projects/graveDetection/tools/train.py projects/graveDetection/configs/test/cascade_rcnn_r50_fpn_1x-area2.py
`
Here is my training log:
[20200128_134005.log](https://github.com/open-mmlab/mmdetection/files/4120904/20200128_134005.log)
I trained it today and it stuck after 1 epoch this time.

Because my images have four bands, so i write a new image loading file('LoadImageFromFile' -> 'LoadMultiBandsImageFromFile', 'Normalize' -> 'NormalizeMultiBands') to fit images with four bands.I use PIL to read images and I did not change the original program logic. At last, the pre-trained model I use is ResNet-50 provided publicly. I just change the first kernel from size [64, 3, 7, 7] to size [64, 4, 7, 7]. These are all my changes. Is it what caused the problem? But when the interval is 1, the training is normal.

**Comment by ZwwWayne** [comment]
Hi @ZR19940115 ,
I tried [mask_rcnn_r50_fpn_1x.py](https://github.com/open-mmlab/mmdetection/blob/master/configs/mask_rcnn_r50_fpn_1x.py) using the newest mmdet by only changing the checkpoint interval=4 and met no bugs, it is been trained at the 8th epoch now. Therefore, the bug seems to be caused by somewhere other than the original code base.

From your log, the program stuck after one epoch training before evaluation, you might need to check your dataloader and check its behavior during evaluation.

**Comment by ZR19940115** [comment]
Well, thank you, I'll check it more carefully
